### PR TITLE
Feat : Added Modal 

### PR DESCRIPTION
**Pull request by srejitk**
This Merge Request adds the following :

1. Modal Component
2. Modal Mini Component
3. Documentation for the above
Modal

### CODE REVIEW

File: docs/Modal.html [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Modal</title>
+    <link rel="stylesheet" href="/main.css">
+    <link rel="stylesheet" href="/stylesheets/components/Modal/Modal.css">
+    <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.4.0/styles/night-owl.min.css"
+        integrity="sha512-i5X6Fdn/ZqvGSqPrdMa3FgcpXM/Nr6YccSKFYT93zljl/HZDEpvBbE5Pxp91eiWGccZLrL/LDQJd7fjTRYsVaA=="
+        crossorigin="anonymous" referrerpolicy="no-referrer" />
```
Comment (nishasen) [inline docs/Modal.html:16]: Its better if you import these files in separate css file and import that link here, your code will look more cleaner

File: docs/Modal.html [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+</head>
+
+<body>
+    <div class="main-container">
+        <!-- Header HTML -->
+        <header id="header" class="iris__header">
+            <nav class="iris__navbar">
+                <div class="iris__navbar__logo__section">
+                    <a class="home__section">
```
Comment (nishasen) [inline docs/Modal.html:25]: add href attribute to a tag

File: stylesheets/components/Modal/Modal.css [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    background-color: var(--white);
+    color: var(--component-green-05);
+    border: 2px solid var(--component-green-05);
+}
+
+.modal-critical {
+    background-color: var(--component-red-05);
+    color: white;
+    border: 2px solid transparent
+}
```
Comment (nishasen) [inline stylesheets/components/Modal/Modal.css:78]: you can create a separate class for modal which will have the common code

**Review by nishasen — APPROVED** [review]
Code is clean, and I like your naming conventions. Suggested some small changes.

**Review by sadathshariff — APPROVED** [review]
Good Naming conventions used and you can work on the Suggestions From Nisha

File: stylesheets/components/Modal/Modal.css (hunk 3 above)
Comment (srejitk) [reply stylesheets/components/Modal/Modal.css:78]: There's no common properties for the Modal, so hence it's all seperate

File: docs/Modal.html (hunk 1 above)
Comment (srejitk) [reply docs/Modal.html:16]: Yes, will correct it in another merge

**Pull request merged**